### DISCUSSION TITLE
What does rerun do?

### DISCUSSION THREAD

**Discussion by cadop** [community/community#26804] [Actions]
<p>I am still learning the yml format and how things work, so maybe this is not the most common of problems, but I am pretty confused on the wording of ‘rerun’.</p>
<p>So essentially what happened was:</p>
<ul>
<li>I have some folder to ignore for pytest. Someone made a PR (1) that changed that folders name but without any other things.  For some reason, none of the github actions ran (I am guessing they first set the PR to a branch that doesnt have actions, then changed the branch to merge into that does have actions, and then it didn’t run)</li>
<li>I merged the PR 1</li>
<li>The next PR (2) was for something unrelated, but now the github action fails because the folder that was being ignored was renamed. it does not change the yml file</li>
<li>I updated the yml file to fix the name</li>
<li>I went back to the PR 2 and clicked rerun, but it fails again, because it is using the old yml file</li>
</ul>
<p>So what is rerun supposed to do? Just rerunning the same thing that will fail will keep failing if it doesn’t include any changes of the repo, right? Or am I not using it correctly?</p>

**Comment by kingthorin** [comment]
<p>It will depend what code is checked out by the workflow. In the scenario you described the second PR would need to be rebased on the updated default branch, and force pushed, so that the proper workflow runs.</p>

**Comment by cadop** [comment]
<p>So workflows run on the branch they come from, but not what the branch will look like if it is merged? That seems so strange, especially if using features like unittests to make sure a change doesn’t break anything.</p>

**Comment by kingthorin** [comment, marked as answer]
<p>What runs (in most cases…there are exceptions) is the workflow in the destination repo (for security reasons) what it runs against depends on what is checked out by the workflow.</p>

**Comment by koffie** [comment]
<p>Hi cadop, The actual code the workflows runs on depends on the trigger that you have configured.</p>
<p>on: [ push ]</p>
<p>will run the workflow on the code that has been pushed to a certain branch</p>
<p>on: [ pull_request ]</p>
<p>will run the workflow on how the code looks after it is merged.</p>

**Comment by cadop** [comment]
<p>Hi Koffie, thanks for trying to help so long after the post.  However, I am not sure this is the full picture.  I have a github action that is running  on  [pull_request], and the same thing as my original post is happening.</p>
<p>Two PRs were supplied at the same time. PR1 was merged first, now PR2 is not updating the check when using “Re-run jobs”.</p>
<p>PR1 → Changes a value in the code, that breaks a unit test. it gets Merged (with failing test). Only a single file is in the ‘Files Changed’<br>
PR2 →  Changes the unit test so it works correctly, this check <em>should</em> be passing.  (the merged version does pass) but it fails, using the new values of the unit test, but has ignored the latest codebase which includes PR1. However the Files changed only shows the unittest update (i.e. its not also reverting the code change).</p>
